tests/ipv6-autoconfig.nix

```diff
@@ -0,0 +1,101 @@
+import ./make-test-python.nix ({ testlib, ... }:
+let
+  makePhysicalHost = { id, links }: { lib, config, ... }:
+    let
+      testNodeId = config.virtualisation.test.nodeNumber;
+    in {
+      imports = [
+        (testlib.fcConfig {
+          inherit id;
+          net.mgm = true;
+          net.ul = true;
+          extraEncParameters = {
+            inherit id;
+            interfaces.fe.policy = "vxlan";
+            interfaces.srv.policy = "vxlan";
+            interfaces.ul = {
+              policy = "underlay";
+              nics = map (link: {
+                mac = "52:54:00:12:${lib.toLower (lib.toHexString link)}:0${toString testNodeId}";
+                external_label = "phys/${toString link}";
+              }) links;
+            };
+          };
+        })
+      ];
+
+      # use the hardware networking config profile
+      flyingcircus.networking.physicalHostNetworking = true;
+      # extra underlay network links
+      virtualisation.vlans = links;
+
+      services.fail2ban.enable = false;
+    };
+in {
+  name = "ipv6-autoconfig";
+  testCases = {
+    virtual = {
+      name = "virtual";
+      nodes.machine = { ... }: {
+        imports = [
+          (testlib.fcConfig {})
+        ];
+      };
+      testScript = ''
+        sysctls = [
+            "accept_ra",
+            "autoconf",
+            "temp_valid_lft",
+            "temp_prefered_lft",
+            "addr_gen_mode",
+        ]
+
+        machine.wait_for_unit("multi-user.target")
+
+        with subtest("testing ipv6 autoconf configuration on ethsrv"):
+            for sysctl in sysctls:
+                machine.succeed(f"sysctl net.ipv6.conf.ethsrv.{sysctl} | grep 0")
```

Comment:
I'm a bit frustrated by the switch from Python to Bash ... ;) Specifically the grep feels like a shotgun.

If we need to stay within bash then this would be a bit more exact: `[ "$(sysctl  net.ipv6.conf.ethsrv.accept_ra -n -b)" == "1" ]` ... BUT ... it's terrible to read IMHO ... 

I'll let you decide.

Reply:
On second thought, the sysctls which handle temporary SLAAC-generated addresses take times in seconds, so this would match if there's a 0 digit in the number of seconds. I think a better solution might be to write a script which asserts that the sysctl is 0, and then invoke that in the test.